iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -201,6 +200,15 @@ public List<InputSplit> getSplits(JobContext context) {
         .ofNullable(HiveIcebergStorageHandler.table(conf, conf.get(InputFormatConfig.TABLE_IDENTIFIER)))
         .orElseGet(() -> Catalogs.loadTable(conf));
 
+    if (conf.get(InputFormatConfig.TABLE_IDENTIFIER) == null) {
+      conf.set(InputFormatConfig.TABLE_IDENTIFIER, table.name());
+    }
+
+    if (conf.get(InputFormatConfig.SERIALIZED_TABLE_PREFIX + conf.get(InputFormatConfig.TABLE_IDENTIFIER)) == null) {
+      conf.set(InputFormatConfig.SERIALIZED_TABLE_PREFIX + conf.get(InputFormatConfig.TABLE_IDENTIFIER),
```

Comment:
IIUC,  every split has a `conf `and each `conf `would contain a srialized iceberg table. I want to know if the srialized iceberg table would make `conf `bigger? and also would cause AM oom in case of many splits?
Did we have a benchmark test to verify the validation of this change?

Reply:
- Conf contained serialized Iceberg table even before these changes and all q-tests besides one were passing without adding serialized table to conf. So these changes actually do not increase memory usage anywhere, but just decrease, because table is removed from the splits.

- One q-test that was failing was show_partitions_test.q. It was failing because it is doing select from _table.**partitions**._ 

- There were 2 unit test classes (not q-tests) that were failing because serialized table wasn't in conf, but it is something that was present only in unit tests.

For these 2 edge cases, I added serialized table into conf in `IcebergInputFormat.getSplits()`.

Reply:
I almost see what you mean. thx